### ISSUE TITLE
only log [Run Config] when running from config

### ISSUE THREAD

**Issue by AdriaandeJongh** [poohcom1/godot-run-configs#4]
I'm getting the "[Run Config] Current config:" output when I run my game with the regular play button, which makes it look like it's running from the config runner, but it's not: the 'main' scene is ran instead of the custom one in the mentioned config. So it seems like incorrect behavior.

**Comment by poohcom1** [comment]
It's due to limitations with Godot addons.

Godot's plugin interface [doesn't offer much](https://docs.godotengine.org/en/stable/classes/class_editorinterface.html#class-editorinterface-method-play-current-scene) in terms of setting up the scene being played, you can pass any kind of data between the editor and game. I need some form of communication with the played game for run configs to work, so my solution was to save the config info into the project settings, which will be read by the autoload when the game starts. 

Because all the data is in the project setting, it comes into effect regardless of how the scene was run. So I treated the run button more like an optional button you can use that will run the custom scene, while the native button still uses the config, just not its set scene.

There's probably some way to make it work by temporarily setting flags in the project setting or to a separate file, but it's all super hacky and error-prone so I decided against it.

**Comment by AdriaandeJongh** [comment]
To clarify: if I run the game using the regular play button, it doesn't run the scene configured in the config, whereas it does when I run using the additional button. So it's not exactly the same behavior, and that's confusing – especially when there's a log entry that makes it seem like it should be.

I don't know what the technical solution here is either! I'm just confused and that confusion is stopping me from taking this plugin to the rest of my team, even though it is almost precisely what I was looking for.

**Comment by poohcom1** [comment]
> To clarify: if I run the game using the regular play button, it doesn't run the scene configured in the config, whereas it does when I run using the additional button.

The config runner has two parts: the environment variables and the custom "Run Scene". The environment variables come into effect regardless of which button you use. The `[Run Config]` log is to notify that the environment variables from a config are applied, not necessary that a scene from the config is used.

The `Run Scene`, as you said, only comes into effect if you use the additional run button due to the limitations I mentioned. I already clarified this behavior in the readme:

> Run Scene: For this to come into effect, you have to run from the custom run button. Using the regular play buttons will use the respective scene of said button

but I understand the confusion because of the `[Run Config]`. Regardless, I made it like this because my main use case for run configs is the environment variables and I just added the `Run Scene` as a bonus feature, so I didn't see an issue when it doesn't always line up with the logged output.

That being said, I don't have much of an option here. Either:
1. I remove the `Run Scene` feature to reduce confusion.
2. I modify the log message to clarify that environment variables are applied, but not necessarily the run scene.

Both of which don't do much IMO. If you want option 2, then you are free to modify this line here:

https://github.com/poohcom1/godot-run-configs/blob/b74c9cc82b1894426731d566f666818bf98ffefd/addons/run-configs/run-config-manager.gd#L26-L28

**Comment by AdriaandeJongh** [comment]
I think being able to run a different scene with the click of a button is a core feature of this addon so I hope you won't remove it. And I think modifying the log message to state that the environment variables were applied does clarify it a little. I've personally changed the print message to: 
```
print("Applied environment variables for '%s' configuration." % config.name)
```

**Comment by poohcom1** [comment]
I'll update the main message and add a warning like this:

![image](https://github.com/poohcom1/godot-run-configs/assets/74857873/664519b0-6ed0-4fb6-9a19-39651b98a908)

It appears when using `Custom Scene` mode and you ran from a different scene since this is the only kind of scene where you can detect a mismatch, and it's probably the main use case for Run Scene anyway.

I'm also considering just removing the `Main Scene` and `Current Scene` options and just leaving `Custom Scene` from the config options since you can already do that with the native buttons. I think this could help make the intention of the button more clear, but I'm not sure. Let me know if you think that's a good idea.

**Comment by AdriaandeJongh** [comment]
Yeah, that's already way better. I'm personally not a fan of the colored output as it 1. hints at the logs being warnings, and 2. really pulls your attention to it which I don't think is really necessary here. Regarding the `Main Scene` and `Current Scene` options: I think you're right but I would even suggest to go further than that: to make the custom scene a mandatory field for each environment. That way the button in the play bar doesn't have to ever be disabled and the functionality and your new warnings make it clear what's happening.
